packages/editor/tests/oas3.spec.ts

```diff
@@ -0,0 +1,254 @@
+          queryParam.valueJsonPointer,
+          newValue
+        );
+
+        shouldBeValidDoc(openApiEditor['doc']);
+
+        jsonPath.query(tree, path)[0].value.should.equal(newValue);
+
+        (
+          openApiEditor['doc'].paths['/pet/findByTags'].get
+            .parameters[0] as OpenAPIV3.ParameterObject
+        ).example.should.equal(newValue);
+      });
+
+      it('should set referenced request body value', async () => {
+        await openApiEditor.setup(source);
+        let tree = openApiEditor.parse();
+
+        const path =
+          '$..children[?(@.path=="/pet/{petId}" && @.method=="PATCH")].parameters[?(@.bodyType)]';
+        const newValue = '{"name":"test"}';
+
+        const bodyNode = jsonPath.query(
+          tree,
+          path
+        )[0] as SpecTreeRequestBodyParam;
+        bodyNode.should.not.haveOwnProperty('value');
+
+        tree = openApiEditor.setParameterValue(
+          bodyNode.valueJsonPointer,
+          newValue
+        );
+
+        shouldBeValidDoc(openApiEditor['doc']);
+
+        jsonPath.query(tree, path)[0].value.should.equal(newValue);
+
+        (
+          openApiEditor['doc'].paths['/pet/{petId}'].patch
+            .requestBody as OpenAPIV3.RequestBodyObject
+        ).content[bodyNode.bodyType].example.should.equal(newValue);
+      });
+    });
+
+    describe('oas v3 nodes remover', () => {
+      it('should remove path node', async () => {
+        await openApiEditor.setup(source);
+        let tree = openApiEditor.parse();
+
+        const path = '$..children[?(@.path=="/pet/{petId}")]';
+        const pathNode = jsonPath.query(tree, path)[0] as SpecTreeNode;
+
+        tree = openApiEditor.removeNode(pathNode.jsonPointer);
+
+        shouldBeValidDoc(openApiEditor['doc']);
+
+        jsonPath.query(tree, path).should.be.empty;
+        openApiEditor['doc'].paths.should.not.haveOwnProperty(path);
+
+        openApiEditor['tree'].should.be.deep.equal(openApiEditor.parse());
+      });
+
+      it('should remove endpoint node', async () => {
+        await openApiEditor.setup(source);
+        let tree = openApiEditor.parse();
+
+        const path =
+          '$..children[?(@.path=="/pet/{petId}" && @.method=="GET")]';
+        const endpointNode = jsonPath.query(tree, path)[0] as SpecTreeNode;
+
+        tree = openApiEditor.removeNode(endpointNode.jsonPointer);
+
+        shouldBeValidDoc(openApiEditor['doc']);
+
+        jsonPath.query(tree, path).should.be.empty;
+
+        openApiEditor['doc'].paths['/pet/{petId}'].should.not.haveOwnProperty(
+          'get'
+        );
+
+        openApiEditor['tree'].should.be.deep.equal(openApiEditor.parse());
+      });
+    });
+
+    describe('oas v3 serialization', () => {
+      it('should serialize yaml into yaml', async () => {
+        await openApiEditor.setup(source);
+        openApiEditor.stringify().should.be.a('string');
+        openApiEditor.stringify().should.match(/^openapi:/);
```

Comment:
Declare a local variable to avoid multiple calls
```suggestion
        const result = openApiEditor.stringify();
        result.should.be.a('string').and.match(/^openapi:/);
```